server/middleware/token.js

```diff
@@ -22,16 +23,25 @@ module.exports = token;
  * Rewrite the url to replace current user literal with the logged in user id
  */
 function rewriteUserLiteral(req, currentUserLiteral) {
-  if (req.accessToken && req.accessToken.userId && currentUserLiteral) {
+  if (!currentUserLiteral) return;
+  const literalRegExp = new RegExp('/' + currentUserLiteral + '(/|$|\\?)', 'g');
+
+  if (req.accessToken && req.accessToken.userId) {
     // Replace /me/ with /current-user-id/
     var urlBeforeRewrite = req.url;
-    req.url = req.url.replace(
-      new RegExp('/' + currentUserLiteral + '(/|$|\\?)', 'g'),
+    req.url = req.url.replace(literalRegExp,
         '/' + req.accessToken.userId + '$1');
+
     if (req.url !== urlBeforeRewrite) {
       debug('req.url has been rewritten from %s to %s', urlBeforeRewrite,
         req.url);
     }
+  } else if (!req.accessToken && literalRegExp.test(req.url)) {
+    req.res.status(401).end();
```

Comment:
You shouldn't be touching the `res` object in this method. Let the error handler middleware decide how to handle the error.

Sorry if my original comment didn't make this clear.